File: codigo/backend/go-backend/models/vendedor.go
package models



type Vendedor struct {
    ID        uint   `gorm:"primaryKey"`
    Nome      string `gorm:"type:varchar(255);not null"`
    Afiliacao string `gorm:"type:varchar(255);not null"`
    LojaID    uint   `gorm:"not null"`
    Loja      Loja   `gorm:"foreignKey:LojaID"`
}

func (Vendedor) TableName() string {
    return "vendedor"
}

func (v *Vendedor) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "id":        v.ID,
        "nome":      v.Nome,
        "afiliacao": v.Afiliacao,
        "loja_id":   v.LojaID,
    }
}

func (v *Vendedor) String() string {
    return "<Vendedor " + v.Nome + ">"
}

File: codigo/backend/sap-ecc-emulator/src/models/vendedor.py
from db import db

class Vendedor(db.Model):
    __tablename__ = 'vendedores'

    id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(255), nullable=False)
    afiliacao = db.Column(db.String(255), nullable=False)
    loja_id = db.Column(db.Integer, db.ForeignKey('lojas.id'), nullable=False)

    loja = db.relationship('Loja', backref='vendedores', lazy=True)

    def to_dict(self):
        return {
            'id': self.id,
            'nome': self.nome,
            'afiliacao': self.afiliacao,
            'loja_id': self.loja_id
        }

    def __repr__(self):
        return f'<Vendedor {self.nome}>'
